src/bases/finitebasis.jl

```diff
@@ -0,0 +1,60 @@
+import Base: length,
+    size
+
+###############
+# FiniteBasis #
+###############
+    # A FiniteBasis repesents the tensor product
+    # of an arbitrary number of fock bases, each of which is truncated 
+    # at a known, finite number of states. For example, imagine the 
+    # following fock bases and their tensor product:
+    #
+    # F_4 => {|0>, |1>, |2>, |3>}
+    # F_3 => {|0>, |1>, |2>}
+    # F_2 => {|0>, |1>} 
+    #
+    # F_4 x F_3 x F_2 => {|0,0,0>, |0,0,1>, |0,1,0>, |0,1,1>...|3,2,1>}
+    # 
+    # Note that the subscript number refers to basis *length*, not max
+    # excitation level.
+    #
+    # This tensor product basis, F_4 x F_3 x F_2, could be represented 
+    # by calling FiniteBasis(4, 3, 2). 
+
+    immutable FiniteBasis{S} <: AbstractBasis{S}
```

Comment:
Yes, I wanted to replace `lens::(Int...)` by `lens::NTuple{N,Int}`, but you are right, in that case `N` needs to be a type parameter of `FiniteBasis`. Would that be problematic? I am just "concerned" that the performance of Tuples is limiting us at some point?